src/common/utils.h

```diff
@@ -983,9 +986,21 @@ inline bool AlignedMemAlloc(void** ptr, size_t size, size_t alignment) {
   if (*ptr == nullptr)
     return false;
 #else
-  int res = posix_memalign(ptr, alignment, size);
-  if (res != 0)
+  int res                       = posix_memalign(ptr, alignment, size);
+#if __linux__
+  constexpr size_t gHugePage2MB = 1 << 21;
+  if (size >= gHugePage2MB) {
```

Comment:
what about in performance point of view
```suggestion
  if (size >= 2 * gHugePage2MB) {
```

Reply:
Of course, we can replace N with some suitable number, but it is not easy! Depending on the processor architecture, there are at least two different huge page size. On the 86_64 architecture: there is available 2MB and 1GB. This `grep pse /proc/cpuinfo | uniq`  if this command returns a string and a string is non-empty; then either 2MB pages are supported or maybe 1GB is supported `grep pdpe1gb /proc/cpuinfo | uniq` (The 1GiB THP is supported only on the processors with the pdpeg1b CPU flag). Well, the default page size is equal to 4KiB ~ in my scenario I had only possibility to check max: ~2MB THP.